zebra-chain/src/block/header.rs

```diff
@@ -65,3 +65,20 @@ pub struct BlockHeader {
     /// The Equihash solution.
     pub solution: EquihashSolution,
 }
+
+impl BlockHeader {
+    /// Returns true if the header is valid based on its `EquihashSolution`
+    pub fn is_valid(&self) -> bool {
```

Comment:
This should probably take the `n` and `k` from some kind of consensus-parameters object right?

Reply:
absolutely, do you want me to open an issue for that?

Reply:
Yeah, I think that we should think about how we want to represent consensus parameters, maybe in an object in `zebra-consensus` or something?

Reply:
https://github.com/ZcashFoundation/zebra/issues/565